examples/callback_on_end.rs

```diff
@@ -0,0 +1,41 @@
+use std::error::Error;
+use std::io::BufReader;
+use std::sync::atomic::{AtomicU32, Ordering};
+use std::sync::Arc;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let stream_handle = rodio::OutputStreamBuilder::open_default_stream()?;
+    let sink = rodio::Sink::connect_new(&stream_handle.mixer());
+
+    let file = std::fs::File::open("assets/music.wav")?;
+    sink.append(rodio::Decoder::new(BufReader::new(file))?);
+
+    // lets increment a number after `music.wav` has played. We are going to use atomics
+    // however you could also use a `Mutex` or send a message through a `std::sync::mpsc`.
+    let playlist_pos = Arc::new(AtomicU32::new(0));
+
+    // The closure needs to own everything it uses. We move a clone of
```

Comment:
The closure has to be `move` because it is executed in the CPAL's thread. Maybe we should mention that.

Reply:
Would that not be explaining too much of the internals of rodio. I chose 'own' since that is a term even rust beginners are familiar.

A middle ground would be mentioning the closure has to be Send and that is why it needs to own the data?

Reply:
Yes, it is always a problem in documentation because to decide on which details to omit you have to assume the level of the readers. I believe that the closure do not have to be Send if playback is not used (e.g. when writing to a file). Maybe adding "which is necessary for playback" would be enough.

Reply:
> I believe that the closure do not have to be Send if playback is not used

Its part of the EmptyCallback callback type https://docs.rs/rodio/0.20.1/rodio/source/struct.EmptyCallback.html#structfield.callback

Reply:
OK, let's keep it this way.